src/features/session_trace/instrument/index.js

```diff
@@ -71,57 +58,23 @@ export class Instrument extends InstrumentBase {
       handle('bstHist', [location.pathname + location.hash, this.startPath, this.time], undefined, FEATURE_NAMES.sessionTrace, thisInstrumentEE)
     })
 
-    if (supportsPerformanceObserver()) {
-      // capture initial resources, in case our observer missed anything
-      handle(BST_RESOURCE, [window.performance.getEntriesByType('resource')], undefined, FEATURE_NAMES.sessionTrace, thisInstrumentEE)
-
-      observeResourceTimings()
-    } else {
-      // collect resource timings once when buffer is full
-      if (window.performance[CRT] && window.performance[ADD_EVENT_LISTENER]) {
-        window.performance.addEventListener(RESOURCE_TIMING_BUFFER_FULL, this.onResourceTimingBufferFull, eventListenerOpts(false))
-      }
-    }
-
-    function observeResourceTimings () {
-      var observer = new PerformanceObserver((list, observer) => { // eslint-disable-line no-undef
-        var entries = list.getEntries()
-
+    let observer
+    try {
+      // Capture initial resources and watch for future ones. Don't defer this given there's a default cap on the number of buffered entries.
+      observer = new PerformanceObserver((list) => { // eslint-disable-line no-undef
```

Comment:
If we are going to use the PO with buffered true, does this still need to be in the instr file?

Reply:
this would also prevent the need from passing it around between inst and agg

Reply:
This way better ensures there's no gap in the resource collected for **heavy** pages, since the buffer can be filled, and overflow won't be captured w/o an observer.